src/libraries/Common/src/Interop/Unix/System.Security.Cryptography.Native/Interop.OpenSsl.cs

```diff
@@ -149,45 +193,116 @@ internal static SafeSslHandle AllocateSslContext(SslProtocols protocols, SafeX50
                     SetSslCertificate(innerContext, certHandle!, certKeyHandle!);
                 }
 
-                if (sslAuthenticationOptions.IsServer && sslAuthenticationOptions.RemoteCertRequired)
+            }
+            catch
+            {
+                innerContext.Dispose();
+                throw;
+            }
+
+            return innerContext;
+        }
+
+        // This essentially wraps SSL* SSL_new()
+        internal static SafeSslHandle AllocateSslHandle(SafeFreeSslCredentials credential, SslAuthenticationOptions sslAuthenticationOptions)
+        {
+            SafeSslHandle? context = null;
```

Comment:
Please use almost any other name for this variable, since it's not an SSL_CTX (and I'm already trying to figure out the difference between sslCtx and innerContext)